tests/core/test_graph.py

```diff
@@ -1001,14 +1005,30 @@ def f():
 @pytest.mark.parametrize(
     "num_nodes,num_edges", [(100, 200), (1000, 5000), (20000, 100000)]
 )
-@pytest.mark.parametrize("force_adj_lists", ["directed", "undirected"])
+@pytest.mark.parametrize(
```

Comment:
In order to capture some of performance of the filtering in the inner loop, I wonder if if it's worth also parameterising this benchmark by the number of node types, e.g.

```python
@pytest.mark.parametrize("n_types", [1, 4, 20])
```

and passing this through to `example_benchmark_graph`.

(Potentially with a `if force_adj_lists in ("directed", "undirected") and n_types != 1: pytest.skip("...")` to avoid spending time on retesting uninteresting things.)